tracer/src/Datadog.Trace/Debugger/Expressions/MethodScopeMembers.cs

```diff
@@ -0,0 +1,51 @@
+// <copyright file="MethodScopeMembers.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+#if !NET461
+using System.Buffers;
+#endif
+
+namespace Datadog.Trace.Debugger.Expressions;
+
+internal class MethodScopeMembers
+{
+    private int _index = 0;
+
+    public MethodScopeMembers(int numberOfLocals, int numberOfArguments)
+    {
+        // 2 for 'return' and 'exception'
+#if NET461
+        Members = new ScopeMember[numberOfLocals + numberOfArguments + 2];
+#else
+        Members = ArrayPool<ScopeMember>.Shared.Rent(numberOfLocals + numberOfArguments + 2);
+#endif
+    }
+
+    public ScopeMember[] Members { get; private set; }
```

Comment:
Is the order in which we store the scope members here "Locals -> Arguments -> Return Value -> Exception"? I'd suggest documenting this with a XML comment.

Reply:
The order is based on what we received from the native side, but I do not rely on any order.